test/async-script-loader-spec.js

```diff
@@ -14,87 +14,146 @@ class MockedComponent extends React.Component {
-      <ComponentWrapper />
+      <ComponentWrapper asyncScriptOnLoad={asyncScriptOnLoadSpy} />
     );
-    assert.equal(hasScript(URL), true);
+
+    assert.equal(hasScript(URL), false, "Url not in document");
+    assert.equal(asyncScriptOnLoadCalled, true, "asyncScriptOnLoad callback called");
     ReactDOM.unmountComponentAtNode(ReactDOM.findDOMNode(instance));
     instance.componentWillUnmount();
     delete window[globalName];
   });
 
   it("should accept a function for scriptURL", () => {
     const URL = "http://example.com/?url=function";
-    const ComponentWrapper = makeAsyncScriptLoader(MockedComponent, () => URL);
+    const ComponentWrapper = makeAsyncScriptLoader(() => URL)(MockedComponent);
     const instance = ReactTestUtils.renderIntoDocument(
       <ComponentWrapper />
     );
-    assert.equal(hasScript(URL), true);
+
+    assert.equal(hasScript(URL), true, "Url in document");
     ReactDOM.unmountComponentAtNode(ReactDOM.findDOMNode(instance));
     instance.componentWillUnmount();
   });
 
   it("should expose functions with scope correctly", (done) => {
-    const ComponentWrapper = makeAsyncScriptLoader(MockedComponent, "http://example.com/", {
+    const ComponentWrapper = makeAsyncScriptLoader("http://example.com/?functions=true", {
       exposeFuncs: ["callsACallback"],
-    });
+    })(MockedComponent);
     const instance = ReactTestUtils.renderIntoDocument(
       <ComponentWrapper />
     );
     instance.callsACallback(done);
   });
+
   it("should not remove tag script on removeOnUnmount option not set", () => {
     const URL = "http://example.com/?removeOnUnmount=notset";
-    const ComponentWrapper = makeAsyncScriptLoader(MockedComponent, URL);
+    const ComponentWrapper = makeAsyncScriptLoader(URL)(MockedComponent);
     const instance = ReactTestUtils.renderIntoDocument(
       <ComponentWrapper />
     );
-    assert.equal(hasScript(URL), true);
-    ReactDOM.unmountComponentAtNode(ReactDOM.findDOMNode(instance));
-    instance.componentWillUnmount();
-    assert.equal(hasScript(URL), true);
+
+    assert.equal(hasScript(URL), true, "Url in document");
+    const unmounted = ReactDOM.unmountComponentAtNode(ReactDOM.findDOMNode(instance).parentNode);
+    assert.equal(unmounted, true, "successfully unmounted");
+    assert.equal(hasScript(URL), true, "Url still in document after unmounting");
   });
+
   it("should remove tag script on removeOnUnmount option set to true", () => {
     const URL = "http://example.com/?removeOnUnmount=true";
-    const ComponentWrapper = makeAsyncScriptLoader(MockedComponent, URL, { removeOnUnmount: true });
+    const ComponentWrapper = makeAsyncScriptLoader(URL, { removeOnUnmount: true })(MockedComponent);
     const instance = ReactTestUtils.renderIntoDocument(
       <ComponentWrapper />
     );
-    assert.equal(hasScript(URL), true);
-    ReactDOM.unmountComponentAtNode(ReactDOM.findDOMNode(instance));
-    instance.componentWillUnmount();
```

Comment:
added `.parentNode` to the `unmountComponentAtNode` call which now correctly removes the mounted component so the components `componentWillUnmount` is called correctly 😺